var/spack/repos/builtin/packages/fenics-dolfinx/package.py

```diff
@@ -31,6 +31,7 @@ class FenicsDolfinx(CMakePackage):
     )
 
     depends_on("cxx", type="build")  # generated
+    depends_on("c", type="build")  # HDF5 dependency requires C in CMake
```

Comment:
Should the dependency coming through the HDF5 package?

Reply:
I don't think so. Indeed, `"c"` is a *build* dependency of HDF5, but that doesn't make it an automatic build dependency of `fenics-dolfinx`. We need it because CMake checks the C compiler when looking for HDF5.